### PR TITLE
Rtpfec example

### PR DESCRIPTION
**Pull request by MathieuDuponchelle**
This depends on https://bugzilla.gnome.org/show_bug.cgi?id=792696

### CODE REVIEW

**Comment by sdroege** [comment]
The 1.10 dependency should be conditional, like in the decodebin example. Many distros still have 1.8 unfortunately and you even update the requirement for all examples :)

**Comment by MathieuDuponchelle** [comment]
> The 1.10 dependency should be conditional, like in the decodebin example. Many distros still have 1.8 unfortunately and you even update the requirement for all examples :)

Not sure why I needed it tbh, just removed it and things still build fine, updated the PR :)

File: examples/src/bin/rtpfecclient.rs [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+}
+
+fn make_fec_decoder (rtpbin: &gst::Element, sess_id: u32) -> Result<gst::Element, Error> {
+    let fecdec = make_element("rtpulpfecdec", None)?;
+    let internal_storage = rtpbin.emit("get-internal-storage", &[&sess_id.to_value()]).unwrap().unwrap();
+
+    fecdec.set_property("storage", &internal_storage.to_value())?;
+    fecdec.set_property("pt", &100u32.to_value())?;
+
+    Ok (fecdec)
```
Comment (sdroege) [inline examples/src/bin/rtpfecclient.rs]: Please run this through rustfmt-nightly :)

File: examples/src/bin/rtpfecclient.rs [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+            eprintln!("Recording to out.mkv");
+        },
+        _ => return Err(Error::from(UsageError(args[0].clone())))
+    }
+
+    src.link(&netsim)?;
+
+    rtpbin.connect("new-storage", false, |values| {
+        let storage = values[1].get::<gst::Element>().expect("Invalid argument");
+        storage.set_property("size-time", &250000000u64.to_value()).unwrap();
```
Comment (sdroege) [inline examples/src/bin/rtpfecclient.rs]: This is looking for some underscores: `250_000_000`

File: examples/src/bin/rtpfecclient.rs [hunk 3]
```diff
@@ -0,0 +1,290 @@
...
+    assert_ne!(ret, gst::StateChangeReturn::Failure);
+
+    while let Some(msg) = bus.timed_pop(gst::CLOCK_TIME_NONE) {
+        use gst::MessageView;
+
+        match msg.view() {
+            MessageView::Eos(..) => break,
+            MessageView::Error(err) => {
+                {
+                    match err.get_details() {
```
Comment (sdroege) [inline examples/src/bin/rtpfecclient.rs]: Hm this should be 1.10 API?

File: examples/src/bin/rtpfecserver.rs [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+
+    while let Some(msg) = bus.timed_pop(gst::CLOCK_TIME_NONE) {
+        use gst::MessageView;
+
+        match msg.view() {
+            MessageView::Eos(..) => break,
+            MessageView::Error(err) => {
+                {
+                    match err.get_details() {
+                        Some(details) if details.get_name() == "error-details" => details
```
Comment (sdroege) [inline examples/src/bin/rtpfecserver.rs]: In both applications there is nothing that would ever have Some details. You don't post any error message with details anywhere

**Comment by sdroege** [comment]
Doing the same example without FEC might also be nice for a as-simple-as-possible rtpbin example

**Pull request closed without merging**